src/js/game/hud/parts/blueprint.js

```diff
@@ -215,10 +216,22 @@ export class Blueprint {
                     clone.components.StaticMapEntity.origin.addInplace(tile);
 
                     root.map.placeStaticEntity(clone);
-                    root.entityMgr.registerEntity(clone);
+
+                    // Registering a belt immediately triggers a recalculation of surrounding belt
+                    // directions, which is no good when not all belts have been placed. To resolve
+                    // this, only register belts after all entities have been placed.
+                    if (!clone.components.Belt) {
+                        root.entityMgr.registerEntity(clone);
+                    } else {
+                        beltsToRegisterLater.push(clone);
+                    }
                     anyPlaced = true;
                 }
             }
+
+            for (let i = 0; i < beltsToRegisterLater.length; i++) {
+                root.entityMgr.registerEntity(beltsToRegisterLater[i]);
```

Comment:
This looks like a solution, however:
In line 209 a return is possible - This means the entities would have been placed on the map but not registered (Which will lead to a crash). Either we make this a hard assertion (`assertAlways`) so this is a crash anyways, or we make sure that we register the already placed entities there

Reply:
I think making line 209 an `assertAlways` makes the most sense here, since I can't think of a scenario where you would want a building would be both replaceable and unremovable.

Reply:
Good idea